packages/rich-text/src/component/use-anchor-ref.js

```diff
@@ -12,19 +12,25 @@ import { getActiveFormat } from '../get-active-format';
 /** @typedef {import('../register-format-type').RichTextFormatType} RichTextFormatType */
 /** @typedef {import('../create').RichTextValue} RichTextValue */
 
+/**
+ * @typedef {Object} VirtualAnchorElement
+ * @property {Function} getBoundingClientRect A function returning a DOMRect
+ * @property {Document} ownerDocument         The element's ownerDocument
+ */
+
 /**
  * This hook, to be used in a format type's Edit component, returns the active
- * element that is formatted, or the selection range if no format is active.
- * The returned value is meant to be used for positioning UI, e.g. by passing it
- * to the `Popover` component.
+ * element that is formatted, or a virtual element for the selection range if
+ * no format is active. The returned value is meant to be used for positioning
+ * UI, e.g. by passing it to the `Popover` component.
  *
  * @param {Object}                 $1          Named parameters.
  * @param {RefObject<HTMLElement>} $1.ref      React ref of the element
  *                                             containing  the editable content.
  * @param {RichTextValue}          $1.value    Value to check for selection.
  * @param {RichTextFormatType}     $1.settings The format type's settings.
  *
- * @return {Element|Range} The active element or selection range.
+ * @return {Element|VirtualAnchorElement|null|undefined} The active element or selection range.
```

Comment:
> I'm guessing the potential `null` comes from the `element.closest()`

Correct, that's why I had added `null` as a potential return value.

Your suggestion definitely makes sense to me, implemented in https://github.com/WordPress/gutenberg/pull/43713/commits/7cea8ce74848b5b06bf0971a66c63966f774aca2

Reply:
Just flagging that, in the base PR, I've made a change to allow `anchor` to be `null` — this was in response to [some feedback received on a related PR](https://github.com/WordPress/gutenberg/pull/43808#issuecomment-1237764676), where it was noted than allowing `null` as a possible value of the `anchor` would result in a better devX (ie. the Popover can handle the `null` case easily, instead of requiring its consumers to pass `undefined` instead of `null`, which is a typical value that `ref`s can assume in React)

With that in mind, I've also pushed a change to this PR, restoring `useAnchorRef`'s return value to include `null` too https://github.com/WordPress/gutenberg/pull/43713/commits/1bc3b66845cfc4460e77502d091cfd44ed52c2cc